internal/controllers/watchdog/controller.go

```diff
@@ -0,0 +1,71 @@
+package watchdog
+
+import (
+	"context"
+	"time"
+
+	apiv1 "github.com/Azure/eno/api/v1"
+	"github.com/Azure/eno/internal/manager"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// watchdogController exposes metrics that track the states of Eno resources relative to the current time.
+// The idea is to identify deadlock states so they can be alerted on.
+type watchdogController struct {
+	client    client.Client
+	threshold time.Duration
+}
+
+func NewController(mgr ctrl.Manager, threshold time.Duration) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&apiv1.Composition{}).
```

Comment:
The lifecycle controller takes care of this for Pods right? Or I guess that's implicit here, if Pods are stuck then the composition will also be stuck

Reply:
Exactly. Ideally the lifecycle controllers gets pods unstuck, but if not then it will be reflected here